### PR TITLE
routine management

### LINKED ISSUE #9: handle invalid teacherId

### PR DESCRIPTION
**Pull request by Prod-IP**
- [x] create lecture api added
- [x] delete api added
- [x] get lecture by teacher id added
- [x] add types under model folder

### CODE REVIEW

File: src/apis/routine.api.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      .isIn([
+        'monday',
+        'tuesday',
+        'wednesday',
+        'thursday',
+        'friday',
+        'saturday',
+      ])
+      .withMessage(
+        'day must be one of monday, tuesday, wednesday, thursday, friday, saturday',
```
Comment (Bijoy-007) [inline src/apis/routine.api.ts]: @Prodip-Kumar-Paul can't we pass a simple message?  _Please provide a valid day i.e. Monday to sunday_

File: src/apis/routine.api.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  createNewLecture,
+);
+
+router.get(
+  '/get_routine_by_teacher_id',
+  validationErrorHandler,
+  isAuth,
+  hasType(['PRINCIPAL', 'SENIOR_TEACHER']),
+  getLectureByTeacherId,
+);
```
Comment (Bijoy-007) [inline src/apis/routine.api.ts:50]: @Prodip-Kumar-Paul What if the principal wants to know the routine of another teacher? IG in that case we need to ask for teacherId to correctly identify the teacher.

File: src/controllers/routine/delete-lecture.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const isPresent = await lecture.findUnique({
+      where: { id: lectureId },
+    });
+    if (!isPresent) {
+      res.status(200);
+      return throwError('Lecture not found');
+    }
+    const deletedLecture = await lecture.delete({
+      where: { id: lectureId },
+    });
```
Comment (Bijoy-007) [inline src/controllers/routine/delete-lecture.ts]: @Prodip-Kumar-Paul Why we are deleting permanently although we have a property called `isDeleted`?

File: src/controllers/routine/get-lectures-by-teacher-id.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const { day } = req.query as unknown as {
+      day: string;
+    };
+
+    const { id, schoolId, type } = req;
+
+    let teacherId = id;
+    type
+      ? 'PRINCIPAL' || 'SENIOR_TEACHER'
+      : (teacherId = req.query.teacherId as string);
```
Comment (Bijoy-007) [inline src/controllers/routine/get-lectures-by-teacher-id.ts]: @Prodip-Kumar-Paul please write down a comment for better understanding

File: src/controllers/routine/new-lecture.controller.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+const createNewLecture = asyncHandler(async (req, res) => {
+  try {
+    const { name } = req.body as Grade;
+    const { subject, day, startTime, endTime } = req.body as Lecture;
+
+    const isPresent = await grade.findUnique({
+      where: { name },
+    });
+    if (isPresent) {
+      res.status(200);
```
Comment (Bijoy-007) [inline src/controllers/routine/new-lecture.controller.ts]: @Prodip-Kumar-Paul IG 200 is a typo

File: src/apis/routine.api.ts (hunk 1 above)
Comment (Prod-IP) [reply src/apis/routine.api.ts]: it's autocomplete bro

File: src/apis/routine.api.ts (hunk 2 above)
Comment (Prod-IP) [reply src/apis/routine.api.ts:50]: yes that's y I'm checking type and taking id from body

File: src/controllers/routine/delete-lecture.ts (hunk 3 above)
Comment (Prod-IP) [reply src/controllers/routine/delete-lecture.ts]: oh forgot will update in a min

File: src/controllers/routine/get-lectures-by-teacher-id.ts (hunk 4 above)
Comment (Prod-IP) [reply src/controllers/routine/get-lectures-by-teacher-id.ts]: will write now onwards

File: src/controllers/routine/new-lecture.controller.ts (hunk 5 above)
Comment (Prod-IP) [reply src/controllers/routine/new-lecture.controller.ts]: update in a min

**Pull request merged**